### PR TITLE
hnerv_qlp

### PR DESCRIPTION
**Pull request by Bucky789**
# submission name:
hnerv_qlp

# upload zipped `archive.zip`
Hosted as a release asset (`curl -L` works, SHA-256 verified):
https://github.com/Bucky789/comma_video_compression_challenge/releases/download/hnerv-qlp-v1/archive.zip

SHA-256 `ebd513903bd598b4d73d699a11c89600bd11747f27aab26737e518096675b813`, 176,525 bytes, single ZIP member `x` (ZIP_STORED).

# report.txt
```
=== Evaluation results over 600 samples ===
  Average PoseNet Distortion: 0.00003000
  Average SegNet Distortion: 0.00056085
  Submission file size: 176,525 bytes
  Original uncompressed size: 37,545,489 bytes
  Compression Rate: 0.00470163
  Final score: 100*segnet_dist + √(10*posenet_dist) + 25*rate = 0.190946
```
(`evaluate.py` prints the score to 2 dp as `0.19`; full precision is `0.190946`.)

# does your submission require gpu for evaluation (inflation)?
no — inflate is CPU-only (device pinned to CPU); deps `torch`, `numpy`, `constriction`.

# did you include the compression script? and want it to be merged?
yes — the full offline encoder is in `submissions/hnerv_qlp/encoder/` (extract → polish → pack). Happy for it to be merged.

# is this submission competitive or innovative? explain why
Both.

**Competitive:** CPU score `0.190946` vs the current #1 (`rhnerv_comma`, `0.191126`) — a margin of `-0.000180`. Local eval reproduces the #1's published metrics to ~1e-5, so the margin is real (~18x that).

| | SegNet | PoseNet | bytes | score |
|---|---|---|---|---|
| #1 `rhnerv_comma` | 0.00056023 | 0.00002943 | 177,136 | 0.191126 |
| **hnerv_qlp** | 0.00056085 | 0.00003000 | 176,525 | **0.190946** |

**Innovative:** every top submission reuses PR #95's decoder and latents; the only per-pair latent tuning attempted so far is PR #101's sidecar (one dimension, a small fixed step table). This submission generalizes that to **quantization-aware gradient descent over all 28 latent dimensions of every pair**, decoder frozen, optimized directly against SegNet/PoseNet through the exact inflate chain. The polished latents reach the #1's distortion **without** its 607-byte sidecar, so the archive is 611 bytes smaller.

# additional comments
No decoder training. Decoder weights (PR #95) and the FEC6 selector (PR #110) are reused bit-for-bit; the ctx container is PR #112's, with the latent section re-encoded and the sidecar dropped. One implementation note that mattered: the SegNet distortion is a discrete argmax, so fp16 and fp32 disagree on borderline pixels — selecting latents on an fp16 scorer overfits the GPU and regresses on the CPU axis (an fp16-selected build scored 0.191955). Evaluation and per-pair selection are done in fp32; training stays fp16 for speed. Full lineage and attribution in `submissions/hnerv_qlp/THIRD_PARTY_NOTICES.md` (PRs #95/#98/#101/#110/#112).

### CODE REVIEW

**Comment by YassineYousfi** [comment]
added to leaderboard, congrats 
the code has too much overlap with code that is already merged though, so we won't merge it unless you refactor to reuse bits and pieces that are already there
feel free to reopen

**Comment by Bucky789** [comment]
I have emailed the email address mentioned as I'm looking for a Full Stack Engineer job.
Will refactor the code so that it can get merged.
Thank You.
